### PR TITLE
Bug: handle single file dependencies with the --image flag 

### LINKED ISSUE #54: Bug: Consistently document and execute update dependencies
When creating a Promise from operator we are presented the following output:
```
Promise generated successfully.
The Operator documents were added as inline dependencies in the Promise Spec.
You can move them to a workflow by running:
	kratix update dependencies --image yourorg/your-image:tag
```

However, the provided command does not actually work as you need to provide the contents of the dependencies to move to the image. In addition, the docs suggested command also fails as the contents must be provided as a directory rather than a flat file:
Image

```
kratix update dependencies operator-bundle.yaml --image yourorg/your-image:tag
generated the promise/configure/dependencies/configure-deps in promise.yaml
Error: fdopendir operator-bundle.yaml: not a directory
Usage:
  kratix update dependencies [flags]

Examples:
 # update promise dependencies with files in 'local-dir'
Kratix update dependencies local-dir/

Flags:
  -d, --dir string     Directory to read Promise from (default ".")
  -h, --help           help for dependencies
  -i, --image string   Name of the image in which to provide the dependencies within Promise Configure workflow.
```

This story will make sure that CLI help text, command output, and docs all align on the use of the `update dependencies` command and also make sure that the ability to use a flat file OR a directory with the command works as expected.

### PR DESCRIPTION
**Pull request by abangser**
Validated with:
```
/Users/abbybangser/dev/syntasso/kratix-cli/bin/kratix init operator-promise cnpg --split \
      --group mygroup.org \
      --kind Pg-database \
      --version v1alpha1 \
      --operator-manifests operator-bundle.yaml \
      --api-schema-from clusters.postgresql.cnpg.io

/Users/abbybangser/dev/syntasso/kratix-cli/bin/kratix update dependencies operator-bundle.yaml --image yourorg/your-image:tag
docker build -t yourorg/your-image:tag workflows/promise/configure/dependencies/configure-deps
```

Resolves #54

### CODE REVIEW

File: cmd/update_dependencies.go [hunk 1]
```diff
@@ -14,19 +15,21 @@ import (
...
+	Use:   "dependencies PATH",
 	Short: "Commands to update promise dependencies",
-	Long:  "Commands to update promise dependencies",
-	Example: ` # update promise dependencies with files in 'local-dir'
-Kratix update dependencies local-dir/ `,
+	Long:  "Commands to update promise dependencies, by default dependencies are stored in the Promise spec.dependencies field",
+	Example: `# update promise dependencies with all files in 'local-dir'
+kratix update dependencies local-dir/
+# update promise dependencies with single file 'local-file'
+kratix update dependencies local-file`,
```
Comment (kirederik) [inline cmd/update_dependencies.go:24]:
```suggestion
kratix update dependencies path/to/dir/

# update promise dependencies with single file
kratix update dependencies path/to/file.yaml`,
```

File: test/update_test.go [hunk 2]
```diff
@@ -380,6 +380,69 @@ var _ = Describe("update", func() {
 							}))
 						})
 					})
+
+					It("can accept a single file for dependencies", func() {
```
Comment (kirederik) [inline test/update_test.go:384]: this is only testing the "single-file" flow when the `--image` flag is passed in, but i assume it works with or without the flag?

also there's a a lot of duplication with the existing test 🤔 i wonder what's the minimum test that would give us the confidence that it works for a single file... or maybe a table test with the directory case? 🤔

Comment (abangser) [reply test/update_test.go:384]: The bug was only when `--image` flag was passed in, hence the focus on that test case.

As for the duplication, we unfortunately have different code paths for the two use cases, so I did just copy over the validations. There is maybe space to do a refactoring of the code paths, but they are basically two different if/then cases so I leaned against it.

Let's touch base because happy to go a step further on the refactoring if it makes sense in this scope.
